server/app/services/cloud/websocket_client.rb

```diff
@@ -129,17 +130,21 @@ def on_open(event)
 
     # @param [Faye::WebSocket::API::Event] event
     def on_message(event)
-      data = MessagePack.unpack(event.data.pack('c*'))
-      if request_message?(data)
-        EM.defer {
-          response = rpc_server.handle_request(data)
-          send_message(MessagePack.dump(response).bytes)
-        }
-      elsif notification_message?(data)
-        EM.defer {
-          rpc_server.handle_notification(data)
-        }
-      end
+        data = MessagePack.unpack(event.data.pack('c*'))
+        if request_message?(data)
+          EM.defer {
+            if leader?
```

Comment:
This should be checked before we unpack event... we can do this later if there are problems with that.